src/main/java/jenkins/plugins/publish_over_ssh/BapSshTransfer.java

```diff
@@ -46,31 +46,34 @@
     private int execTimeout;
     private boolean usePty;
     private boolean useAgentForwarding;
+    private boolean useSftpForExec;
 
     BapSshTransfer(final String sourceFiles, final String remoteDirectory, final String removePrefix,
                    final boolean remoteDirectorySDF, final boolean flatten, final String execCommand, final int execTimeout) {
-        this(sourceFiles, null, remoteDirectory, removePrefix, remoteDirectorySDF, flatten, execCommand, execTimeout, false, false, false, null);
+        this(sourceFiles, null, remoteDirectory, removePrefix, remoteDirectorySDF, flatten, execCommand, execTimeout, false, false, false, null, false);
     }
 
     public BapSshTransfer(final String sourceFiles, final String excludes, final String remoteDirectory, final String removePrefix,
                           final boolean remoteDirectorySDF, final boolean flatten, final String execCommand, final int execTimeout,
-                          final boolean usePty, final boolean noDefaultExcludes, final boolean makeEmptyDirs, final String patternSeparator) {
+                          final boolean usePty, final boolean noDefaultExcludes, final boolean makeEmptyDirs, final String patternSeparator, final boolean useSftpForExec) {
```

Comment:
Adding to an existing constructor is not a good idea. Please add a setter and annotate it with @DataBoundSetter